### PR TITLE
Add support for /distros endpoint

### PR DESCRIPTION
**Pull request by dbradf**

### CODE REVIEW

File: tests/evergreen/test_util.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from datetime import datetime
+
+import evergreen.util as under_test
+
+SAMPLE_DATE_STRING = "2019-02-13T14:55:37.000Z"
+
+
+class TestParseEvergreenDateTime:
```
Comment (tychoish) [inline tests/evergreen/test_util.py]: shouldn't these be new style classes?

File: tests/evergreen/data/distro.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{
+    "name": "amazon1-2018-build",
+    "user_spawn_allowed": true,
+    "provider": "ec2-auto",
```
Comment (tychoish) [inline tests/evergreen/data/distro.json]: the provider determines (to some extent) the distro settings.... which might be an issue at some point?

File: tests/evergreen/test_util.py (hunk 1 above)
Comment (dbradf) [reply tests/evergreen/test_util.py]: It doesn't really really matter. The class is just used to hold a collection of tests. I can switch it.

File: tests/evergreen/data/distro.json (hunk 2 above)
Comment (dbradf) [reply tests/evergreen/data/distro.json]: I might stop by and ask you about this today.

File: tests/evergreen/test_util.py (hunk 1 above)
Comment (tychoish) [reply tests/evergreen/test_util.py]: Up to you really. I thought there were python3 implications for old-style classes.

**Comment by dbradf** [comment]
evergreen merge

**Pull request merged**